Framework/DataHandling/src/LoadCSNSNexus.cpp

```diff
@@ -0,0 +1,568 @@
+
+  m_file->openGroup("csns", "NXentry");
+  m_file->openGroup("histogram_data", "NXcollection");
+  std::map<std::string, std::string> entries = m_file->getEntries();
+  for (auto j = inputList.begin(); j != inputList.end(); ++j) {
+    auto it = entries.find(*j);
+    if (it != entries.end()) {
+      m_file->openGroup(it->first, it->second);
+      m_file->readData("histogram_data", tmp);
+      data.insert(data.end(), tmp.begin(), tmp.end());
+      m_file->closeGroup();
+    }
+  }
+  m_file->closeGroup();
+  m_file->closeGroup();
+  return data;
+}
+
+/**
+ * load histogram data into workspace
+ * @param[in] workspace ::
+ * @param[in] timeOfFlight ::
+ * @param[in] pidNums :: total pixel numbers
+ * @param[in] histData :: histogram data
+ */
+void LoadCSNSNexus::loadHistData(MatrixWorkspace_sptr &workspace,
+                                 std::vector<uint32_t> &timeOfFlight,
+                                 size_t pidNums,
+                                 std::vector<uint32_t> &histData) {
+  size_t timeNums = timeOfFlight.size();
+  MantidVecPtr x, e;
+  MantidVec xA;
+  MantidVec xRef(timeNums);
+
+  int st1 = 0;
+  for (auto it = timeOfFlight.begin(); it != timeOfFlight.end(); it++) {
+    xRef[st1] = (*it);
+    st1++;
+  }
+
+  std::vector<double> err;
+  for (auto it = histData.begin(); it != histData.end(); it++)
+    err.push_back(sqrt(*it));
+
+  auto it_start = histData.begin();
+  auto it_end = it_start + timeNums - 1;
+
+  auto it_err_start = err.begin();
+  auto it_err_end = it_err_start + timeNums - 1;
+
+  size_t hist = 0;
+  while (hist < pidNums) {
+    auto &Y = workspace->mutableY(hist);
+    auto &E = workspace->mutableE(hist);
+    Y.assign(it_start, it_end);
+    E.assign(it_err_start, it_err_end);
+    it_start += (timeNums - 1);
+    it_end += (timeNums - 1);
+    it_err_start += (timeNums - 1);
+    it_err_end += (timeNums - 1);
+    auto toPass = Kernel::make_cow<HistogramData::HistogramX>(xRef);
+    workspace->setX(hist, toPass);
+    ++hist;
+  }
+}
+
+/**
+ * Get event data
+ * @param[in] inputList :: moduleList or monitorList
+ * @param[in] startList :: list of T0 start number
+ * @param[in] endList :: list of T0 end number
+ * @param[in] pids :: total pixel id
+ * @return event data :: specNo, tof, t0
+ */
+std::multimap<uint32_t, std::pair<float, int64_t>> LoadCSNSNexus::getEventData(
+    const vector<string> &inputList, const vector<uint32_t> &startList,
+    const vector<uint32_t> &endList, const vector<int64_t> &pids) {
+  std::vector<int64_t> pid_tmp;
+  std::vector<int64_t> pidList;
+  std::vector<int64_t> t0_tmp;
+  std::vector<int64_t> t0List;
+  std::vector<float> tof_tmp;
+  std::vector<float> tofList;
+  std::pair<float, int64_t> tofPulse;
+  m_file->openGroup(m_entry, "NXentry");
+  m_file->openGroup("event_data", "NXcollection");
+  std::map<std::string, std::string> entries = m_file->getEntries();
+
+  for (auto j = inputList.begin(); j != inputList.end(); j++) {
```

Comment:
Can I suggest
```c++
for(const auto & name : inputList)
```
as single-letter variable names are better limited to looping over array element indexes.